cmds/core/tty/tty.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"syscall"
+)
+
+func run(stdin io.Writer) error {
+	fi, err := os.Stdin.Stat()
+	if err != nil {
+		return err
+	}
+
+	s, ok := fi.Sys().(*syscall.Stat_t)
```

Comment:
This is a tad confusing. You can stat stdin, but the extended info is not available. Also, can you wrap os.ErrNotExist here? Something like
fmt.Errorf("stat stdin:Sys() is empty:%w", os.ErrNotExist)
for testing?

Reply:
done, but can't come up with ideas how to use it in tests. Any other tests ideas are welcome!